gto/registry.py

```diff
@@ -513,7 +513,7 @@ def check_ref(self, ref: str):
             if aname == name
             for event in artifact.get_events()
             # TODO: support matching the shortened commit hashes
-            if event.ref == ref
+            if event.ref == tag_name
```

Comment:
`event.ref` is poorly named and probably needs to be reworked. For some events it is a tag name (and not a ref) and for others it is a commit SHA.

When we want to compare `refs/...` refs we should be explicitly comparing against a full ref, when we want to compare named tags we should be explicitly comparing against `event.tag` and when we want to compare commit shas we should be explicitly comparing `event.commit_hexsha`